### PR TITLE
Fibonacci class complete w/ unit tests

### PR DESCRIPTION
**Pull request by ryan-b-writin**
hooray!

### CODE REVIEW

**Comment by ryan-b-writin** [comment]
@Odigene12 
@jcockhren 
@katerebekah 
@BlaiseGratton

File: Mathmagician.Tests/FibonacciTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Assert.IsNotNull(fibToCheck);
+        }
+        [TestMethod]
+        public void WhatHappensIfYouEnterZero()
+        {
+            Fibonacci fibWithZero = new Fibonacci();
+            fibWithZero.GenerateFibonacci(0);
+            //the array already contains the first two numbers, 1 1
+            //this assert makes sure no new numbers were added.
+            Assert.IsTrue(fibWithZero.sequence.Count == 2);
```
Comment (katerebekah) [inline Mathmagician.Tests/FibonacciTests.cs]: Are you going to print those two numbers if the user wants 0 numbers??

File: Mathmagician.Tests/FibonacciTests.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        public void WhatAboutALargeNumber()
+        {
+            Fibonacci bigNumber = new Fibonacci();
+            bigNumber.GenerateFibonacci(123451);
+            int third_last_fibonacci = bigNumber.sequence[123448];
+            int second_last_fibonacci = bigNumber.sequence[123449];
+            int last_fibonacci = bigNumber.sequence[123450];
+            Assert.IsTrue(last_fibonacci == (third_last_fibonacci + second_last_fibonacci));
+        }
+    }
```
Comment (katerebekah) [inline Mathmagician.Tests/FibonacciTests.cs:94]: You need a test that does a CollectionAssert to verify the whole list is what you are expecting it to be.

File: Mathmagician.Tests/FibonacciTests.cs (hunk 1 above)
Comment (ryan-b-writin) [inline Mathmagician.Tests/FibonacciTests.cs]: No, those numbers will not print. If 0 is fed into the method, it prints out a string and closes.

Comment (katerebekah) [inline Mathmagician.Tests/FibonacciTests.cs]: What about 1?

File: Mathmagician.Tests/FibonacciTests.cs (hunk 2 above)
Comment (ryan-b-writin) [inline Mathmagician.Tests/FibonacciTests.cs:94]: ok! That's in now :)

File: Mathmagician.Tests/FibonacciTests.cs (hunk 1 above)
Comment (ryan-b-writin) [inline Mathmagician.Tests/FibonacciTests.cs]: I got that taken care of with another if statement to check if the user asks for less than 3.
if - the user asks for 0, it prints a string and closes
else if -the user asks for less than 3, it prints '1' once or twice depending
else- it prints 1 twice, then it starts making fibzos and printing those, adding the new ones to the list.

I went back in and edited the fibonacci class to be more straightforward. Also added a test to make sure that inputting '1' gets you '1' back

**Comment by jcockhren** [comment]
@ryan-b-writin Remove all calls to the `Console` class. Only the Program.cs file should be using the console.

File: Mathmagician/Fibonacci.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        public List<int> Fibonaccis = new List<int>();
+
+        //this method takes an int argument.
+        //if the number of requested fibonaccis is less than 3,
+        //it simply prints off the first or first two numbers in the 'sequence' list.
+        //ask for more than 3, and it will add 1 and 1 to the list, 
+        //start generatin' fibbos, add the generated numbers to the list, rinse & repeat 
+        public void GenerateFibonacci(int numOfFibonacci)
+        {
+            if (numOfFibonacci == 0 )
```
Comment (katerebekah) [inline Mathmagician/Fibonacci.cs:24]: This could be simplified.

Comment (katerebekah) [inline Mathmagician/Fibonacci.cs:24]: You should return a list out of this method and have something else print to the console.  This method is doing too much.

Comment (katerebekah) [inline Mathmagician/Fibonacci.cs:24]: That way you won't need the if statement for the numOfFibonacci

Comment (katerebekah) [inline Mathmagician/Fibonacci.cs:24]: (it'll simplify your tests, too)

**Comment by jcockhren** [comment]
Resolve your conflicts. then :+1:

**Comment by Odigene12** [comment]
Agreed :+1:

**Pull request merged**